lightning/src/routing/router.rs

```diff
@@ -373,11 +373,14 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
 	if final_value_msat == 0 {
 		return Err(LightningError{err: "Cannot send a payment of 0 msat".to_owned(), action: ErrorAction::IgnoreError});
 	}
+	
+	let last_hops = last_hops.clone().iter().map(|hops| hops.0.clone()).collect::<Vec<_>>();
```

Comment:
Clipppy says this (I'm not sure why you added the clone at all - we should be able to just iterate directly and maybe drop this line entirely).

```
error: using `clone` on a double-reference; this will copy the reference instead of cloning the inner type
   --> lightning/src/routing/router.rs:377:18
    |
377 |     let last_hops = last_hops.clone().iter().map(|hops| hops.0.clone()).collect::<Vec<_>>();
    |                     ^^^^^^^^^^^^^^^^^
    |
    = note: `#[deny(clippy::clone_double_ref)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_double_ref
help: try dereferencing it
```